spec.html

```diff
@@ -32,8 +32,118 @@ <h1>Well-known intrinsic objects</h1>
 	</emu-table>
 </emu-clause>
 
+<emu-clause id="sec-wrapped-function-exotic-objects">
+	<h1>Wrapped Function Exotic Objects</h1>
+	<p>A wrapped function exotic object is an exotic object that wraps a callable object. A wrapped function exotic object is callable (it has a [[Call]] internal method). Calling a wrapped function exotic object generally results in a call of its wrapped function.</p>
+
+	<p>An object is a <dfn id="wrapped-function-exotic-object">wrapped function exotic object</dfn> if its [[Call]] internal method use the following implementations, and its other essential internal methods use the definitions found in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>. These methods are installed in WrappedFunctionCreate.</p>
+
+	<p>Wrapped function exotic objects do not have the internal slots of ECMAScript function objects listed in <emu-xref href="#table-internal-slots-of-ecmascript-function-objects"></emu-xref>. Instead they have the internal slots listed in <emu-xref href="#table-internal-slots-of-wrapped-function-exotic-objects"></emu-xref>, in addition to [[Prototype]] and [[Extensible]].</p>
+	<emu-table id="table-internal-slots-of-wrapped-function-exotic-objects" caption="Internal Slots of Wrapped Function Exotic Objects" oldids="table-28">
+	<table>
+		<tbody>
+		<tr>
+		<th>
+			Internal Slot
+		</th>
+		<th>
+			Type
+		</th>
+		<th>
+			Description
+		</th>
+		</tr>
+		<tr>
+		<td>
+			[[WrappedTargetFunction]]
+		</td>
+		<td>
+			Callable Object
+		</td>
+		<td>
+			The wrapped function object.
+		</td>
+		</tr>
+		<tr>
+		<td>
+			[[Realm]]
+		</td>
+		<td>
+			Realm Record
+		</td>
+		<td>
+			The realm in which the wrapped function object was created.
+		</td>
+		</tr>
+		</tbody>
+	</table>
+	</emu-table>
+
+	<emu-clause id="sec-wrapped-function-exotic-objects-call-thisargument-argumentslist">
+	<h1>[[Call]] ( _thisArgument_, _argumentsList_ )</h1>
+	<p>The [[Call]] internal method of a wrapped function exotic object _F_ takes arguments _thisArgument_ (an ECMAScript language value) and _argumentsList_ (a List of ECMAScript language values). It performs the following steps when called:</p>
+	<emu-alg>
+		1. Let _target_ be _F_.[[WrappedTargetFunction]].
+		1. Assert: IsCallable(_target_) is *true*.
+		1. Let _targetRealm_ be ? GetFunctionRealm(_target_).
+		1. Let _callerRealm_ be ? GetFunctionRealm(_F_).
+		1. For each element _key_ of _argumentsList_, do
+			1. Let o be _argumentsList_[_key_].
+			1. Set _argumentsList_[_key_] to ? GetWrappedValue(_targetRealm_, _o_).
+		1. Let _wrappedThisArgument_ to ? GetWrappedValue(_targetRealm_, _thisArgument_).
+		1. Let _value_ be ? Call(_target_, _wrappedThisArgument_, _argumentsList_).
```

Comment:
`? Call` will leak error objects from another realm.

```js
// other.js
export function x() { null.a }
// main.js
try {
     new Realm().importBinding("./other.js", "x")()
} catch(e) {
    e instanceof TypeError // but in another realm
}
```

Reply:
thanks for pointing this out.

Reply:
fixed.